lib/src/compiler-path.ts

```diff
@@ -4,20 +4,22 @@
 
 import * as fs from 'fs';
 import * as p from 'path';
+import {getELFInterpreter} from './elf';
 import {isErrnoException} from './utils';
 
 /**
- * Detect if the current running node binary is linked with musl libc by
- * checking if the binary contains a string like "/.../ld-musl-$ARCH.so"
+ * Detect if the given binary is linked with musl libc by checking if
+ * the interpreter basename starts with "ld-musl-"
  */
-const isLinuxMusl = function () {
-  return fs.readFileSync(process.execPath).includes('/ld-musl-');
+const isLinuxMusl = function (path: string): boolean {
+  const interpreter = getELFInterpreter(path);
```

Comment:
I wonder if we should wrap this in a `try`/`catch`, print any error it produces to `stderr`, and just fall back to returning false if it errors. If the user is running musl, running the executable will error anyway, but if they're not that'll allow them to continue working without being strictly blocked on debugging why their ELF format isn't what we expect.